### PR TITLE
Prototype/decals gpu optimization

### PR DESCRIPTION
**Pull request by Paulm-Unity**

### CODE REVIEW

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs [hunk 1]
```diff
@@ -109,16 +113,32 @@ public RenderTargetIdentifier[] GetDBuffers()
             return m_ColorMRTs;
         }
 
-		public void ClearNormalTarget(Color clearColor, CommandBuffer cmd)
+		public void ClearNormalTargetAndHTile(Color clearColor, CommandBuffer cmd)
 		{
 			// index 1 is normals
 			CoreUtils.SetRenderTarget(cmd, m_ColorMRTs[1], ClearFlag.Color, clearColor);
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipeline.cs]: Why just clearing normal ? We must clear all of them

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl [hunk 2]
```diff
@@ -34,5 +34,22 @@ void Frag(  PackedVaryingsToPS packedInput,
 	float3x3 decalToWorld = (float3x3)UNITY_ACCESS_INSTANCED_PROP(matrix, normalToWorld);
     GetSurfaceData(positionDS.xz, decalToWorld, surfaceData);
 
+	// have to do explicit test since compiler behavior is not defined for RW resources and discard instructions
+	if((all(positionDS.xyz > 0.0f) && all(1.0f - positionDS.xyz > 0.0f)))
+	{
+		uint mask = 0;
+#if _COLORMAP
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl]: This must not be here: _COLORMAP, _NORMALMAP,_MASKMAP are dependent of DecalData (think about using a shader graph to author the decal). Maybe our GetSurfaceData could return the mask instead.

**Comment by sebastienlagarde** [comment]
I Merged the PR but we will update the code next PR

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl (hunk 2 above)
Comment (Paulm-Unity) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl]: Yes, I see your point and agree, will fix next PR.

**Comment by robbiesri** [comment]
Late to the party, but changes made sense.

**Pull request merged**